nxcomp/src/Loop.cpp

```diff
@@ -3810,7 +3810,11 @@ int SetupAuthInstance()
 
         launchdAddrUnix.sun_family = AF_UNIX;
 
+#if defined(__FreeBSD_kernel__) && defined(__GLIBC__)
```

Comment:
Hardcode this and the other definition to `104`. All BSD variants I could find (FreeBSD, NetBSD, OpenBSD, Dragonfly) and even OS X use a size of `104`.

Debian/kFreeBSD is a bit different in that it also defines `sun_path` as `104` chars long, but appends another internal `4` chars wide field for padding/Linux compatibility. Despite that, the comment `sun_path` comment claims that the kernel can only handle `104` chars.

Reply:
I guess NoMachine just looked up the structure on their Linux-based development machines when writing that code and never actually bothered to check the target OS headers...

Reply:
@Ionic: the code passage in discussion comes from me, not from NoMachine.

I will look at the code and possibly to the hard-coding to 104 chars.

Reply:
@Ionic: oh, actually only the code in Proxy.cpp comes from me, the code in Loop.cpp is NoMachine'ish, I guess...

Reply:
No, not really. According to `git blame`, both hunks came from NoMachine.